jax/_src/lax/control_flow.py

```diff
@@ -2049,6 +2049,8 @@ def custom_root(f, initial_guess, solve, tangent_solve):
       - For vector ``y``, you could use a linear solve with the Jacobian, if
         dimensionality of ``y`` is not too large:
         ``lambda g, y: np.linalg.solve(jacobian(g)(y), y)``.
+    has_aux: bool indicating whether the ``solve`` and ``tangent_solve`` functions
+      return auxiliary data like solver diagnostics as a second argument.
```

Comment:
I don't see a good reason for including `has_aux` in `tangent_solve`, given that the aux values from tangent solve are always discarded. People might be surprised by how they are always replaced by zeros.

Reply:
Makes sense, I took it out and changed the docstring to mention only `solve`. 

As a small counterpoint: What do you think about giving users the option of outputting the aux values instead of symbolic zero tangents, e.g. by setting a flag?
It would technically break the JVP contract (which is why it's not great) but in turn, it would not inadvertently destroy the information from tangent_solve (and also `solve` in `custom_linear_solve`'s JVP, I guess), and we're operating on constants here anyhow, strictly speaking.

Reply:
> As a small counterpoint: What do you think about giving users the option of outputting the aux values instead of symbolic zero tangents, e.g. by setting a flag?
> It would technically break the JVP contract (which is why it's not great) but in turn, it would not inadvertently destroy the information from tangent_solve (and also `solve` in `custom_linear_solve`'s JVP, I guess), and we're operating on constants here anyhow, strictly speaking.

Sure, in theory we could do this, which would allow for extracting aux information from explicit JVP calls.

But do we really care enough about forward mode autodiff for this to be worth the trouble? I'm not sure. There's no way we can use this trick for reverse mode, unfortunately.

Reply:
Good point. I guess in the tangent vs cotangent space thinking, currently the aux values would naturally be interpreted as tangents, since they are missing in reverse mode. Unless some higher-level aux handling machinery gets introduced, this is probably the most natural solution.